### PR TITLE
On interior vehicle data notification

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Implementation of OnInteriorVehicleDataNotification

### CODE REVIEW

File: src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  const Json::Value module_type =
+      module_data.get(message_params::kModuleType, Json::Value(""));
+
+  return module_type.asString();
+}
+
+std::vector<std::string> OnInteriorVehicleDataNotification::ControlData(
+    const Json::Value& message) {
+  Json::Value data =
+      message.get(message_params::kModuleData, Json::Value(Json::objectValue));
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc]: @LuxoftAKutsan const

File: src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  std::string module = ModuleType(message);
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc]: @LuxoftAKutsan const

File: src/components/interfaces/MOBILE_API.xml [hunk 3]
```diff
@@ -5808,6 +5809,12 @@
         <description> true if successful; false, if failed </description>
     </param>
   </function>
+
+  <function name="OnInteriorVehicleData" functionID="OnInteriorVehicleDataID" messagetype="notification">
+    <param name="moduleData" type="ModuleData">
```
Comment (OlesiaV) [inline src/components/interfaces/MOBILE_API.xml]: please add mandatory="true"

Comment (AByzhynar) [reply src/components/interfaces/MOBILE_API.xml]: @OlesiaV If it's absent - that means parameter has mandatory= true

**Review by LitvinenkoIra — APPROVED** [review]

File: src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h [hunk 4]
```diff
@@ -30,38 +30,51 @@
...
+  virtual ~OnInteriorVehicleDataNotification();
+
+ protected:
+  std::string ModuleType(const Json::Value& message) FINAL;
+  std::vector<std::string> ControlData(const Json::Value& message) FINAL;
 
- private:
-  DISALLOW_COPY_AND_ASSIGN(OnInteriorVehicleDataNotificationValidator);
+ protected:
+  bool Validate() OVERRIDE {
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h]: @LuxoftAKutsan why do you need overriden `Validate()` here? This notifications should not be validated through API schema?

File: src/components/interfaces/MOBILE_API.xml (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/interfaces/MOBILE_API.xml]: @AByzhynar we are using API from https://github.com/yang1070/sdl_core/blob/master/src/components/interfaces/MOBILE_API.xml
  ANd mandatory=true is there

**Review by OlesiaV — COMMENTED** [review]
Didn't find 
param name="isSubscribed" type="Boolean" mandatory="false"  for GetInteriorVehicleData response, will it be added in some other PR?

File: src/components/interfaces/MOBILE_API.xml (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/interfaces/MOBILE_API.xml]: @OlesiaV added

File: src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      application_manager::MessagePtr message =
+          utils::MakeShared<application_manager::Message>(*msg);
+      message->set_message_type(
+          application_manager::MessageType::kNotification);
+      message->set_protocol_version(application_manager::kV3);
+      message->set_function_id(functional_modules::ON_INTERIOR_VEHICLE_DATA);
+      message->set_function_name(MessageHelper::GetMobileAPIName(
+          functional_modules::ON_INTERIOR_VEHICLE_DATA));
+      message->set_connection_key(app.app_id());
+      NotifyOneApplication(message);
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc:90]: @LuxoftAKutsan probably you also should include `moduleData` parameter to this message as it's present in response

**Review by AKalinich-Luxoft — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h (hunk 4 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h]: @AKalinich-Luxoft  this is pure virtual function, and should be overrided. If some notificaiton will need additional validaiotn rules (not according schema), we will need this virtual method

File: src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h [hunk 6]
```diff
@@ -30,38 +30,51 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_VALIDATORS_ON_INTERIOR_VEHICLE_DATA_NOTIFICATION_VALIDATOR_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_VALIDATORS_ON_INTERIOR_VEHICLE_DATA_NOTIFICATION_VALIDATOR_H_
+#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_COMMANDS_ON_INTERIOR_VEHICLE_DATA_NOTIFICATION_H_
+#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_COMMANDS_ON_INTERIOR_VEHICLE_DATA_NOTIFICATION_H_
 
-#include "can_cooperation/validators/validator.h"
-#include "utils/macro.h"
```
Comment (AByzhynar) [inline src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h]: @LuxoftAKutsan Probably this lilne should be returned back as `FINAL` used in this translation unit

File: src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc (hunk 5 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc:90]: @AKalinich-Luxoft it copy all parameters from response

**Review by OlesiaV — APPROVED** [review]

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h (hunk 6 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/include/can_cooperation/commands/on_interior_vehicle_data_notification.h]: @AByzhynar f2fd469

**Review by AByzhynar — APPROVED** [review]

**Review by AKalinich-Luxoft — APPROVED** [review]

**Pull request merged**